src/layouts/_components/UserMenu.js

```diff
@@ -17,41 +17,59 @@ const menuItems = [
       { to: "/user/referral", text: "Free Credits" },
       {
         to: "https://www.stormkit.io/docs",
-        text: "Docs"
+        text: "Docs",
       },
-      { to: "/logout", text: "Logout" }
-    ]
+      { to: "/logout", text: "Logout" },
+    ],
   },
   {
     title: "Get in touch",
     children: [
       {
         to: "https://discord.gg/6yQWhyY",
-        text: "Discord"
+        text: "Discord",
       },
-      { to: "mailto:hello@stormkit.io", text: "Email" }
-    ]
+      { to: "mailto:hello@stormkit.io", text: "Email" },
+    ],
   },
   {
     title: "Legal",
     children: [
       {
         to: "https://www.stormkit.io/policies/privacy",
-        text: "Privacy Policy"
+        text: "Privacy Policy",
       },
       {
         to: "https://www.stormkit.io/policies/terms",
-        text: "Terms of Service"
-      }
-    ]
-  }
+        text: "Terms of Service",
+      },
+    ],
+  },
 ];
 
 const UserMenu = () => {
   const [isMenuOpen, setIsMenuOpen] = useState(false);
 
+  const navRef = useRef(null);
+
+  const handleClickOutside = (event) => {
```

Comment:
@filippofinke have you seen the `src/components/OutsideClick/OutsideClick.tsx` component? Perhaps we can leverage that one here :)

Reply:
Okay, now it should be good :D